explainers/hidden-content-explainer.md

```diff
@@ -86,181 +87,142 @@ currently possible.
-   This mitigation will prevent the page from building out a string of what the
-   user is searching for without having to reveal that string to the user
-   visually. Although this is already possible as shown in 
-   [search-incremental.html](/resources/find-in-page/search-incremental.html),
-   firing beforematch on hidden content that stays hidden would be harder for
-   the user to notice.
-
-As for the ScrollToTextFragment case, the page can already guess with reasonable
-certainty based on the scroll offset which text is already highlighted. In
-addition, ScrollToTextFragment can only occur once at the beginning of a
-document's load, so we don't have to worry about the page building out a search
-term since there will only be one possible beforematch event.
+In order to mitigate find-in-page snooping from both `scroll` events and
+`hidden=until-found`/`beforematch`, browsers should avoid scrolling to every
+character the user types into the find-in-page dialog by adding a delay between
+keystrokes and scrolling. This prevents the page from incrementally building the
+user's search query one character at a time. If a snooping page has to build the
+user's search query at something like 5 characters at a time, the page would
+need to include so many potential results that it would become very hard or
+impossible to recreate the user's search query.
+
+As for the ScrollToTextFragment case, the browser firing the `beforematch` event
+and removing the `hidden` attribute on the target text does not reveal more
+information than is already revealed by the exising scrolling and `:target`
+pseudo selector that ScrollToTextFragment applies.
 
 ## Responses to DOM and style changes in the `beforematch` event handler
 
-The beforematch event handler, as well as any other script that runs during the
-async steps before we actually scroll to the target match, affects the outcome
-of the scroll. As mentioned in the privacy section, if the DOM range
-representing the active match is collapsed or it has a style which makes it
-invisible, the scroll will be canceled. Otherwise, the target match will be
-scrolled into view.
+If the `beforematch` event handler
+[collapses](https://dom.spec.whatwg.org/#range-collapsed) the DOM range of the
+target text or changes the style to remove the target text from rendering, the
+browser will not scroll to the target text.
+
+## Usage of `content-visibility`
+
+`hidden=until-found` will be implemented as a presentational style of
+`content-visibility:hidden` as opposed to the existing `hidden` attribute which
+applies a presentational style of `display:none`. This will improve the
+performance of searching the content inside the `hidden=until-found` subtree
+because searching for text in these areas requires computing style and layout.
+`content-visibility` saves this computed style and layout instead of
+`display:none` which deletes them, which means that once the first run of
```

Comment:
```suggestion
because `content-visibility: hidden`, unlike `display: none`, allows the rendering state
needed to perform a text search to be cached.
This means that once the first run of
```